common/lib/xmodule/xmodule/modulestore/__init__.py

```diff
@@ -270,14 +270,22 @@ def _end_bulk_operation(self, structure_key, emit_signals=True):
         if bulk_ops_record.active:
             return
 
-        dirty = self._end_outermost_bulk_operation(bulk_ops_record, structure_key)
+        # The bulk op has ended. However, the signal tasks below still need to use the
+        # built-up bulk op information (if the signals trigger tasks in the same thread).
```

Comment:
@doctoryes I am curious, if in production, the triggered tasks do not use the built-up bulk information, why should they use it if they are being triggered in the same thread? Wouldn't that create a difference in behaviour between production and test runs?

This slowness could be caused by this task which loads the whole course without doing its own `bulk_operations()`: https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/content/course_structures/tasks.py#L12. Perhaps this was masked earlier because the signals were being trigged inside the parent `bulk_operations()`?